src/EFCore.SqlServer/Query/Internal/Translators/SqlServerTimeOnlyMethodTranslator.cs

```diff
@@ -62,7 +62,8 @@ public SqlServerTimeOnlyMethodTranslator(ISqlExpressionFactory sqlExpressionFact
 
         if ((method == FromDateTime || method == FromTimeSpan)
             && instance is null
-            && arguments.Count == 1)
+            && arguments.Count == 1
+            && arguments[0].Type != typeof(TimeSpan))
         {
             return _sqlExpressionFactory.Convert(arguments[0], typeof(TimeOnly));
```

Comment:
I believe you should always return here when the method is matched. Something like this might work:

```suggestion
            && arguments.Count == 1)
        {
            return arguments[0].Type is typeof(TimeSpan)
                ? arguments[0]
                : _sqlExpressionFactory.Convert(arguments[0], typeof(TimeOnly));
```